src/r/recordFactories/ProjectFactory.ts

```diff
@@ -320,6 +326,26 @@ export class ProjectFactory extends RecordFactory<RT.project> {
     }
   }
 
+  duplicateSceneDeepRecord<N extends RT>(this: ProjectFactory, sceneId: number, type: N, id: number): RecordNode<N> | undefined {
+    const sceneJson = this.getRecord(RT.scene, sceneId);
+    if (sceneJson !== undefined) {
+      const sceneF = r.scene(sceneJson);
+      const duplicatedRecord = sceneF.duplicateDeepRecord(type, id);
+      this.addLinkedVariables([ duplicatedRecord as RecordNode<N> ]);
+      return duplicatedRecord;
+    }
+  }
+
+  duplicateDeepRecord<N extends RT>(this: ProjectFactory, type: N, id: number): RecordNode<N> | undefined {
+    const duplicatedScene = super.duplicateDeepRecord(type, id);
```

Comment:
this check isn't correct.

duplicateDeepRecord can be called for any record types that are children of a project (menu, tour_mode, scenes, etc)

Correct check:
```
const duplicatedRecord = super.duplicateDeepRecord();
switch(type) {
  case RT.scene: {
    const sceneF = r.scene(duplicatedScene);
      const elements = sceneF.getAllDeepChildrenWithFilter(RT.element, e => e.props.element_type === en.ElementType.embed_html); // add condition for other relevant element types.
      this.addLinkedVariables(elements);
  }
}
```

Reply:
Add more test cases to cover all possible scenarios. This wasn't tested properly.